src/pgduckdb.cpp

```diff
@@ -122,53 +135,54 @@ DuckdbInitGUC() {
 	                     &duckdb_log_pg_explain);
 
 	DefineCustomVariable("duckdb.enable_external_access", "Allow the DuckDB to access external state.",
-	                     &duckdb_enable_external_access, PGC_SUSET);
+	                     &duckdb_enable_external_access, PGC_SUSET, 0, GucCheckDuckDBNotInitdHook);
```

Comment:
nit: does it make sense to add another template without the `flags` argument, so we don't need to specify `0` soo many times.

Reply:
Ah I went back and forth on that. I thought I remembered that you created those with all the possible args, so didn't want to add toe many helpers.

But maybe we do want a `DefineCustomDuckDBVariable` that would do exactly that. I'll give it a shot so we can compare